x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -303,147 +310,137 @@ func (a *Aggregator) ProcessBatch(ctx context.Context, b *model.Batch) error {
 //     can aggregate over. Once this limit is breached the metrics will be
 //     aggregated in the `_other` transaction bucket of a dedicated service
 //     with `service.name` as `_other`.
-func (a *Aggregator) AggregateTransaction(event model.APMEvent) {
+func (a *Aggregator) AggregateTransaction(event model.APMEvent) error {
 	count := event.Transaction.RepresentativeCount
 	if count <= 0 {
-		return
+		return nil
 	}
 	for _, interval := range a.Intervals {
-		key := a.makeTransactionAggregationKey(event, interval)
-		a.updateTransactionMetrics(key, count, event.Event.Duration, interval)
+		key := makeTransactionAggregationKey(event, interval)
+		if err := a.updateTransactionMetrics(key, count, event.Event.Duration, interval); err != nil {
+			return err
```

Comment:
IIANM, the error would only ever occur if there's a bug in our code, and not because of something wrong with the data. Right? In that case, should we just log the error rather than returning it?

Reply:
Actually we're not even doing anything with the error, it's being ignored in ProcessBatch. Should we log it there?

Reply:
> Actually we're not even doing anything with the error, it's being ignored in ProcessBatch. Should we log it there?

Ah, this was a miss, thanks for catching it. I planned to add a log entry in `ProcessBatch` and also return the error from there.

> IIANM, the error would only ever occur if there's a bug in our code, and not because of something wrong with the data. Right?

Yes, this is true.

> In that case, should we just log the error rather than returning it?

I was thinking that if there is a bug the process batch should indicate this to the client rather than only logging it. This would give the error more visibility. I don't have a strong preference either way, what do you prefer?